compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -366,6 +411,128 @@ void LinalgFusePass::runOnOperation() {
     return signalPassFailure();
   }
 }
+namespace {
+/// Pattern to wrap splitReduction transformation. This changes the
+/// lowering_config for the split op and keeps the same lowering_config for the
+/// reduction op.
+struct CodegenSplitReduction
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  CodegenSplitReduction(MLIRContext *context, int64_t size,
+                        uint64_t unrollFactor,
+                        linalg::LinalgTransformationFilter filter,
+                        PatternBenefit benefit = 1)
+      : OpInterfaceRewritePattern<linalg::LinalgOp>(context, benefit),
+        size(size),
+        unrollFactor(unrollFactor),
+        filter(std::move(filter)) {}
+
+  LogicalResult matchAndRewrite(linalg::LinalgOp op,
+                                PatternRewriter &rewriter) const override {
+    // Make sure that only generic ops are supported, only one input and the
+    // innermost tensor dimension is reduction dimension.  Everything else is
+    // copied from the original SplitReduction Transformation.
+    if (failed(filter.checkAndNotify(rewriter, op)) ||
+        !op.hasTensorSemantics() || op.getNumReductionLoops() != 1 ||
+        op.getNumOutputs() != 1 || !op.hasOnlyProjectedPermutations() ||
+        !isa<linalg::GenericOp>(op) || op.getNumInputs() != 1)
+      return rewriter.notifyMatchFailure(op, "precondition not met");
+    SmallVector<unsigned> dims;
+    op.getReductionDims(dims);
+    AffineMap map = op.getTiedIndexingMap(op.getInputOperand(0));
+    unsigned lastIdx = map.getNumResults() - 1;
+    unsigned lastDim = map.getDimPosition(lastIdx);
+    if (lastDim != dims[0])
+      return rewriter.notifyMatchFailure(op, "precondition not met");
+
+    linalg::ControlSplitReductionFn fn = [this, lastIdx](linalg::LinalgOp op) {
+      return linalg::SplitReductionOptions{size, lastIdx, true};
+    };
+
+    auto opNumLoops = op.getNumLoops();
```

Comment:
Maybe declare the variable before `lastIdx`, then we don't need `lastIdx` variable. It can be computed through `opNumLoops - 1`.

Also, I think it's not promising that the input operand contains all the dimensions in result map right?

[optional] Lets use `numLoops` like other naming in the file.

Reply:
`opNumLoops` and `lastIdx` have nothing to do with each other. `opNumLoops` specifies the number of loops, and `lastIdx` is 1 less than the number of output dimensions.

> Also, I think it's not promising that the input operand contains all the dimensions in result map right?

Not sure what you mean here.

> [optional] Lets use numLoops like other naming in the file.
Fixed.